pkg/services/accesscontrol/resourcepermissions/api.go

```diff
@@ -430,3 +435,13 @@ func permissionSetResponse(cmd setPermissionCommand) response.Response {
 	}
 	return response.Success(message)
 }
+
+func MiddlewareReceiverUIDResolver(paramName string) web.Handler {
+	return func(c *contextmodel.ReqContext) {
+		gotParams := web.Params(c.Req)
+		if uid, ok := gotParams[paramName]; ok {
+			gotParams[paramName] = alertingac.ScopeReceiversProvider.GetResourceIDFromUID(uid)
+			web.SetURLParams(c.Req, gotParams)
+		}
+	}
+}
```

Comment:
I wonder if this middleware should be in `ngalert/accesscontrol`. This will follow the approach used above, in the case of `teams` resources.